File: array/moneyCalc.java
package springIntro;

import java.util.*;

public class moneyCalc {

	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		double money = 0;
		int cash[] = { 200, 100, 50, 20, 10, 5, 1 };
		int freqC[] = new int[7];
		double penny[] = { 0.50, 0.25, 0.1, 0.05 };
		int freqP[] = new int[4];
		System.out.println("Enter a money");
		money = in.nextFloat();
		int i = 0;
		while (i < 7) {
			int counter = (int) money / cash[i];
			freqC[i] = counter;
			money -= (cash[i] * freqC[i]);
			i++;
		}
		i = 0;
		money += 0.001;
		while(i < 4) {
			int counter =  (int) (money / penny[i]);
			freqP[i] = counter;
			money -= (penny[i] * freqP[i]);
			i++;
		}
		for(i = 0; i < 7; i++) {
			System.out.printf("%d piece %d\n", freqC[i], cash[i]);
		}
		for(i = 0; i < 4; i++) {
			System.out.printf("%d piece %f\n", freqP[i], penny[i]);
		}
	}
}

File: array/moneyCalc.cpp
#include<iostream>
using namespace std;
int main(){
	float money = 0;
	int freqC[7] = {0};
	int freqP[4] = {0};
	int cash[7] = {200, 100, 50, 20, 10, 5, 1};
	float penny[4] = {0.5, 0.25, 0.1, 0.05};
	cout << "Enter a money = ";
	cin >> money;
	while(money >= 1){
		for(int i = 0; i < 7 && money >= 1; i++){
			money -= cash[i];
			freqC[i]++;
			if(money < 0){
				money += cash[i];
				freqC[i]--;
			}
		}
	}
	while(money > 0.1){
		for(int i = 0; i < 4 && money != 0.001; i++){
			money -= penny[i];
			freqP[i]++;
			if(money < 0.0){
				money += penny[i];
				freqP[i]--;
			}
		}
	}
	for(int i = 0; i < 7; i++){
		cout << freqC[i] << " pieces of " << cash[i] << endl;
	}
	for(int i = 0; i < 4; i++){
		cout << freqP[i] << " pieces of " << penny[i] << endl;
	}
	
	return 0;
}
